json-schema-form.d.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Shorthand to lookup for keys with `x-jsf-*` preffix.
+ */
+export function pickXKey(node: Object, key: 'presentation' | 'errorMessage'): Object | undefined;
+
+type JSFConfig = {
+  initialValues?: Record<string, unknown>;
```

Comment:
docs: They are missing, I've added them. Also, the `errorMessage` type can be improved. All is here:

```js

type ValidationTypes = 'type' | 'minimum' | 'maximum' | 'minLength' | 'maxLength' | 'pattern' | 'maxFileSize' | 'accept';

type JSFConfig = {
  /**
   * Initial json values to prefill the form fields.
   * This influences the initial visibility of conditional fields.
  */
  initialValues?: Record<string, unknown>;
  /**
   * It inforces all json properties (fields) to have x-jsf-presentation.inputType.
    @default true 
  */
  strictInputType?: boolean;
  /**
   * Customize the output of a given named Field.
   * Useful if you want to enhance the UI/UX of a particular field.
   * @example
   * { has_pet: { "optionsDirection": "horizontal" } }
  */
  customProperties?: {
    description: (description: string, field: $TSFixMe) => string | string;
    [key: string]: unknown;
  };
  /**
   * Customize the config for each inputType
  */
 inputTypes?: {
    /**
     * Customize the error error message of this input type.
     * @example
     * { required: "This number is required." }
     */
    errorMessage?: Record<ValidationTypes, string>;
  };
};
```

Reply:
P.S. The `Record<ValidationTypes, string>` seems incorrect, what I mean is that each key must one of one the ValidationTypes. #LearningTS

Reply:
I have a couple of doubts here:

* if we give `inputTypes.errorMessage`  the value of` { required: "This number is required." }`, it'll fail as required is not present in the `ValidationTypes`.

* After reviewing the JSFConfig type, should the description property inside customProperties be optional instead of required?

Reply:
- I forgot to add `required` to `ValidationTypes` 🤦 
- You are right, `description` is optional too yes! Basically every key of `JSFConfig` is optional.

Reply:
Fixed it

Reply:
Hum, I believe something is still wrong because this is valid but TS marks it as invalid:

```js
const {fields, handleValidation} = createHeadlessForm(demo, {
    inputTypes: {
      errorMessage: {
        required: 'sss',
      },
    },
  });
```

<img width="639" alt="image" src="https://github.com/remoteoss/json-schema-form/assets/14869087/a5a32216-8a00-400f-b299-a86607f8247a">

Reply:
We were telling typescript that we needed an object with all the `ValidationTypes` keys.

Fixed that with `Partial`

Reply:
TIL about `Partial`. it's perfect now!